src/components/Common/UI/Heading/Heading.test.tsx

```diff
@@ -0,0 +1,88 @@
+import { describe, expect, it } from 'vitest'
+import { screen } from '@testing-library/react'
+import { Heading } from './Heading'
+import { renderWithProviders } from '@/test-utils/renderWithProviders'
+
+describe('Heading', () => {
+  it('renders with correct heading level', () => {
+    renderWithProviders(<Heading as="h2">Test Heading</Heading>)
+    expect(screen.getByRole('heading', { level: 2 })).toBeInTheDocument()
+    expect(screen.getByText('Test Heading')).toBeInTheDocument()
+  })
+
+  it('renders with default h1', () => {
+    renderWithProviders(<Heading as="h1">Default Heading</Heading>)
+    expect(screen.getByRole('heading', { level: 1 })).toBeInTheDocument()
+  })
+
+  it('applies custom className', () => {
+    const { container } = renderWithProviders(
+      <Heading as="h2" className="custom-heading">
+        Styled Heading
+      </Heading>,
+    )
+    expect(container.querySelector('.custom-heading')).toBeInTheDocument()
+  })
+
+  it('renders with different heading levels', () => {
+    const { rerender } = renderWithProviders(<Heading as="h1">Main Heading</Heading>)
+    expect(screen.getByRole('heading', { level: 1 })).toBeInTheDocument()
+
+    rerender(<Heading as="h3">Sub Heading</Heading>)
+    expect(screen.getByRole('heading', { level: 3 })).toBeInTheDocument()
+  })
```

Comment:
Nit: this one feels a little redundant since we're already testing this in the first test